From: Christophe Fergeau <cfergeau@redhat.com>
Date: Sun, 20 Dec 2015 20:51:52 +0100
Subject: [PATCH] NetworkAgent: Fix double-unref in get_secrets_keyring_cb()

In get_secrets_keyring_cb, we own a ref on the 'attributes' hash table
from secret_item_get_attributes), and a ref on the 'secret' object (from
secret_item_get_secret(), but in the SHELL_KEYRING_SK_TAG case, we unref
these once before breaking out of the loop, and the second time after
breaking out of the loop.

https://bugzilla.gnome.org/show_bug.cgi?id=759708

(cherry-picked from commit 021cecbce289f1d9c68a156d5b8dd204a73bc715)
---
 src/shell-network-agent.c | 2 --
 1 file changed, 2 deletions(-)

Index: gnome-shell-3.18.3/src/shell-network-agent.c
===================================================================
--- gnome-shell-3.18.3.orig/src/shell-network-agent.c	2016-01-31 15:14:29.470801330 +0100
+++ gnome-shell-3.18.3/src/shell-network-agent.c	2016-01-31 15:14:29.466801245 +0100
@@ -315,8 +315,6 @@
 
               secrets_found = TRUE;
 
-              g_hash_table_unref (attributes);
-              secret_value_unref (secret);
               break;
             }
         }
